src/overlays/actors/ovl_En_Gs/z_en_gs.c

```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
+}
+
+void func_809984F4(EnGs* this, GlobalContext* globalCtx) {
+    EnGs* gossipStone = NULL;
+
+    do {
+        gossipStone = (EnGs*)func_ActorCategoryIterateById(globalCtx, &gossipStone->actor, ACTORCAT_PROP, ACTOR_EN_GS);
+        if (gossipStone != NULL) {
+            if ((this != gossipStone) && (this->unk_194 == gossipStone->unk_194)) {
+                gossipStone->unk_19A |= 1;
+                func_80999AC0(gossipStone);
+                gossipStone->unk_19C = 3;
+                func_80998704(gossipStone, globalCtx);
+            }
+            gossipStone = (EnGs*)gossipStone->actor.next;
+        }
+    } while (gossipStone != NULL);
+
+    func_800B7298(globalCtx, &this->actor, 7);
+    this->actionFunc = func_809985B8;
+}
+
+void func_809985B8(EnGs* this, GlobalContext* globalCtx) {
+    EnGs* gossipStone;
+    Vec3f sp38;
+
+    if (func_8013E2D4(&this->actor, this->unk_212, -1, 0)) {
+        Player* player = GET_PLAYER(globalCtx);
+
+        Matrix_RotateY(this->actor.shape.rot.y, MTXMODE_NEW);
+        Matrix_GetStateTranslationAndScaledZ(160.0f, &sp38);
+        Math_Vec3f_Sum(&player->actor.world.pos, &sp38, &player->actor.world.pos);
+        Math_Vec3f_Copy(&player->actor.prevPos, &player->actor.world.pos);
+        this->unk_200 = 0.0f;
+        gSaveContext.roomInf[126][1] =
+            (gSaveContext.roomInf[126][1] & ~(7 << (this->unk_198 * 3))) | ((this->unk_194 & 7) << (this->unk_198 * 3));
+        gossipStone = NULL;
+
+        do {
+            gossipStone =
+                (EnGs*)func_ActorCategoryIterateById(globalCtx, &gossipStone->actor, ACTORCAT_PROP, ACTOR_EN_GS);
+            if (gossipStone != NULL) {
+                if ((gossipStone != this) && (gossipStone->actor.params == ENGS_2) &&
+                    (gossipStone->unk_198 == this->unk_198)) {
+                    gossipStone->unk_194 = this->unk_194;
+                    gossipStone->unk_200 = 0.0f;
+                }
+                gossipStone = (EnGs*)gossipStone->actor.next;
+            }
+        } while (gossipStone != NULL);
+
+        func_80998704(this, globalCtx);
+    }
+}
+
+void func_80998704(EnGs* this, GlobalContext* globalCtx) {
+    this->unk_19D = 0;
+    this->unk_19A &= ~(0x100 | 0x4);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EV_G_STONE_CHANGE_COLOR);
+    this->actionFunc = func_8099874C;
+}
+
+void func_8099874C(EnGs* this, GlobalContext* globalCtx) {
+    u32 phi_v0 = false;
+
+    switch (this->unk_19C) {
+        case 0:
+            phi_v0 = func_80998A48(this, globalCtx);
+            break;
+
+        case 1:
+            phi_v0 = func_80998BBC(this, globalCtx);
+            break;
+
+        case 2:
+            phi_v0 = func_80998D44(this, globalCtx);
+            break;
+
+        case 3:
+        case 5:
+            phi_v0 = func_80998F9C(this, globalCtx);
+            break;
+
+        case 4:
+            phi_v0 = func_809995A4(this, globalCtx);
+            break;
+    }
+
+    if (!phi_v0) {
```

Comment:
Looks like this variable isn't actually boolean, at least at this point, from what some of those functions output; does `== 0` match? Also, some of those returns are negative; can it be `s32`?